### ISSUE TITLE
main: reference to nonexistent package os

### ISSUE THREAD

**Issue by jamesbond3142** [golang/go#50509]
### What version of Go are you using (`go version`)?
Go 1.17.6 (originally reproducible in Go 1.17.5, I upgraded to GO 1.17.6 in the hope it is fixed, but it's not)

<pre>
sandbox# go version
go version go1.17.6 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
sandbox# go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/root/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/root/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/go-1.17.6"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/opt/go-1.17.6/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.6"
GCCGO="/usr/bin/gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/root/hello/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3390657096=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
Followed the Go tutorial from Go website: https://go.dev/doc/tutorial/getting-started
I went as far as step 5 (creating hello.go).
When I tried to do step 6 (run go program: `go run .`), I encounter the error.

### What did you expect to see?
The hello.go program to run and show the text "Hello world!" on console.

### What did you see instead?
The following error:

<pre>
# example/hello
2022/01/08 15:48:06 main: reference to nonexistent package os
sandbox# 
</pre>

**Comment by seankhliao** [comment]
Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and proposals only.

For questions please refer to https://github.com/golang/go/wiki/Questions

**Comment by jamesbond3142** [comment]
Wow. 

First and foremost, **I am not asking questions.**

1. I downloaded Go from **official** distribution.
2. I followed installation method from the **official** Go website.
3. I followed steps from the **official** Go tutorial documentation.
4. I get an **error** message compiling a "hello world" program.

Elsewhere in the world, this would be considered as a **product defect**.
Either int the software, or documentation, or both. 
And that's what I'm reporting.

And your first response is to close this issue, because you think I'm just **asking questions**?

Unbelievable.

**Comment by ALTree** [comment]
It does seems unlikely that a Hello world (and thus probably every Go program in existence) was broken in go1.17 and nobody noticed before. My guess is that either you made a mistake when following the tutorial steps, or (more likely) your Go installation is broken in some way.

The only other instance (applicable to go 1.17) of that error is in #48565. In that issue, the OP messed with the `.a` files of the Go standard library, and thus every Go program was broken on importing these files.

If fact, if I run `strip os.a` and then try to `go run` an hello world, I get the same error:

```
$ gotip run test.go
# command-line-arguments
2022/01/08 11:47:30 main: reference to nonexistent package os
```

Did you do something like that? Or in general, you could try to nuke your installation and re-install Go.

**Comment by jamesbond3142** [comment]
Thank you for the more sympathetic response. Really appreciate it.

> It does seems unlikely that a Hello world (and thus probably every Go program in existence) was broken in go1.17 and nobody noticed before. 

Indeed. Which was why I was concerned enough to raise the ticket.

> My guess is that either you made a mistake when following the tutorial steps, or (more likely) your Go installation is broken in some way.
> 
> The only other instance (applicable to go 1.17) of that error is in #48565. In that issue, the OP messed with the `.a` files of the Go standard library, and thus every Go program was broken on importing these files.

I saw the error message shown by #48565 when I was attempting to build a much more complex app. I thought it was the app's problem, but to help to isolate the problem, I tried the hello world program - and was surprised that it was similarly affected. 

However, not having enough knowledge about Go's internal and external modules, I failed to see the connection between the similar errors (In other languages, "non-existent reference" at link time meant missing shared libraries etc, but I found it hard-to-believe that a core library like "os" could be missing). Hence my concern.

> 
> In fact, if I run `strip os.a` and then try to `go run` a hello world, I get the same error:
> 
> ```
> $ gotip run test.go
> # command-line-arguments
> 2022/01/08 11:47:30 main: reference to nonexistent package os
> ```
> 
> Did you do something like that? Or in general, you could try to nuke your installation and re-install Go.

Thank you. This gives me the clue that I need. 

My Go installation is broken. The OS package management automatically strip .a files, assuming that they are standard libraries, behind my back. After re-installing and making sure it does not do this, the compilation process works normally and the hello world program works.

Perhaps a suggestion for future version is not to use .a file extension because of its existing meaning. 

Most importantly, however, **thank you** @ALTree for spending the time and effort to reply to a bug report properly. We need more people like you.
